cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -662,14 +683,18 @@ private List<ApplicationId> decodeAndValidateBatchApplicationRecord(NamespaceId
           throw new BadRequestException("Missing 'name' in the request element for app-id.");
         }
         if (element.getAppVersion() == null) {
-          appIds.add(validateApplicationId(namespaceId, element.getName()));
+          // fetch the latest version
+          String latestAppVersion = getLatestAppVersion(validateApplicationId(namespaceId, element.getName()));
+          appIds.add(validateApplicationVersionId(namespaceId, element.getName(), latestAppVersion));
         } else {
           appIds.add(validateApplicationVersionId(namespaceId, element.getName(), element.getAppVersion()));
         }
       }
       return appIds;
     } catch (JsonSyntaxException e) {
       throw new BadRequestException("Request body is invalid json: " + e.getMessage());
+    } catch (Exception e) {
```

Comment:
Again here, should be specific on what exception this can throw. General exception should not be translated to specific exception thrown

Reply:
changed this as per the discussion here https://github.com/cdapio/cdap/pull/14627/files#r996174596